### ISSUE TITLE
IDE0019 false positive (Use pattern matching)

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#18053]
**Version Used**: Visual Studio 2017 15.0.0-RTW+26228.4

**Code to Reproduce**:
```csharp
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp.Syntax;

public static class C
{
    public static void M(ParameterSyntax parameter)
    {
        SyntaxNode parent = parameter.Parent as BaseParameterListSyntax;

        if (parent != null)
        {
            parent = parent.Parent;
        }
    }
}
```
**Actual Behavior**:
```csharp
public static class C
{
    public static void M(ParameterSyntax parameter)
    {
        if (parameter.Parent is BaseParameterListSyntax parent)
        {
            parent = parent.Parent; // CS0266
        }
    }
}
```
